hw/xbox/mcpx_apu.c

```diff
@@ -1612,6 +1721,21 @@ static void mcpx_apu_register(void)
 }
 type_init(mcpx_apu_register);
 
+#if USE_APU_THREAD
+static void *mcpx_apu_frame_thread(void *arg)
+{
+    volatile MCPXAPUState *d = MCPX_APU_DEVICE(arg);
+
+    while (!d->exiting) {
+        int should_run = (d->regs[NV_PAPU_SECTL] & NV_PAPU_SECTL_XCNTMODE) >> 3;
+        if (should_run == NV_PAPU_SECTL_XCNTMODE_OFF) continue; // FIXME: use a proper sync method
+        se_frame((void*)d);
```

Comment:
There should at least be a warning somewhere in `se_frame`, that this runs at the wrong frequency (host dependent, because it depends on the FIFO being empty).

Reply:
`se_frame` will run at ~1500Hz, give or take a little. This is throttled by the system audio sink. Agree, that could be more clear here, so I'll add a comment about it in Audio PR#2.